ESP/I2C_ESP32/main/main.cpp

```diff
@@ -1,201 +1,17 @@
-        } else if(ret == ESP_OK) {
-			std::cout << "Task[" << task_idx << "] master write:\n";
-            master.printBuffer(master.getTxBuffer(), master.getRwBufferLength());
-        } else {
-			std::cout << "Task [" << task_idx << "] " << esp_err_to_name(ret) << ": Master write slave error, IO not connected...\n";
-        }
-
-		ret = master.read();
-        
-        if(ret == ESP_ERR_TIMEOUT) {
-			std::cout << "I2C TIME_OUT\n";
-        } else if(ret == ESP_OK) {
-			std::cout << "TASK[" << task_idx << "] master read:\n";
-            master.printBuffer(master.getRxBuffer(), master.getRwBufferLength());
-        } else {
-			std::cout << esp_err_to_name(ret) << ": Master read slave error, IO not connected...\n";
-        }
-
-        vTaskDelay(( delayTimeBetweenItemsMS * ( task_idx + 1 ) ) / portTICK_RATE_MS);
-    }
+void app_main();
 }
 
-static void slave(void * taskID) {
-
-	i2c_port_t slavePortNum = i2c_port_t::I2C_NUM_0;	///< Slave port number
-	gpio_num_t slaveSDA = gpio_num_t::GPIO_NUM_25;		///< GPIO number for slave dataBuffer
-	gpio_num_t slaveSCL = gpio_num_t::GPIO_NUM_26;		///< GPIO number for slave CLK
-
-	I2cEsp slave(slaveSDA, slaveSCL, slavePortNum);
-
-    uint32_t task_idx = (uint32_t) taskID;
-	
-    int count = 0;
-	int size = 0;
-	
-	size_t d_size = 0;
-	
-    while(1) {
-		std::cout << "==================\n";
-		std::cout << "test count: " << count++ << "\n";
-		std::cout << "==================\n";
+void app_main() {
+    ///< masterAndSlave demo code -- uncomment to activate
+    AccessBuffers accessBuffers;
 
-		size = slave.read();
-		
-        if(size == ESP_ERR_TIMEOUT) {
-			std::cout << "I2C TIME_OUT\n";
-        } else {
-			std::cout << "Task[" << task_idx << "] slave read: [" << size << "] bytes:\n";
-            slave.printBuffer(slave.getDataBuffer(), size);
-		}
-		
-		std::cout << "MODIFYING DATA...\n";
-		slave.modifyData();
-		
-		d_size = slave.write(slave.getDataBuffer());
+    // xTaskCreate(accessBuffers.printMasterSlaveBuffer, "masterAndSlave", 1024 * 2, (void *)1, 10, NULL);
 
-        if(d_size == 0) {
-			std::cout << "slave transmission buffer is FULL!\n";
-        } else {
-			
-		std::cout << "TASK[" << task_idx << "] Slave write:\n";
-		slave.printBuffer(slave.getDataBuffer(), d_size);
-		}
-			
-        vTaskDelay(( delayTimeBetweenItemsMS * ( task_idx + 1 ) ) / portTICK_RATE_MS);
-    }
-}
-
-void app_main() {
-	///< masterAndSlave demo code -- uncomment to activate
-    xTaskCreate(masterAndSlave, "masterAndSlave", 1024 * 2, (void* ) 1, 10, NULL);
-	
-	///< master - slave demo code -- uncomment to activate
-    ///< xTaskCreate(master, "master", 1024 * 2, (void* ) 1, 10, NULL);
-    ///< xTaskCreate(slave, "slave", 1024 * 2, (void* ) 1, 10, NULL);
+    ///< master - slave demo code -- uncomment to activate
+    /// xTaskCreate(master, "master", 1024 * 2, (void *)1, 10, NULL);
+    xTaskCreate(accessBuffers.masterPrintBuffer, "master", 1024 * 2, (void *)1, 10, NULL);
```

Comment:
This looks like example code that sould be placed on the wiki. That will make your main as clean as possible.